frontend/src/features/settings/account/EmailAddressForm.tsx

```diff
@@ -1,19 +1,68 @@
 import React, { useState, useEffect } from 'react';
 
-import Button from '@material-ui/core/Button';
-import Grid from '@material-ui/core/Grid';
-import { CircularProgress, Box, Typography } from '@material-ui/core';
+import {
+  CircularProgress,
+  Box,
+  Typography,
+  Theme,
+  Grid,
+  Button,
+} from '@material-ui/core';
+import { Link as RouterLink } from 'react-router-dom';
 
 import { useSnackbar } from 'notistack';
 
 import { showErrorAlert } from '../../../utils/notifications';
 import { FormTextField } from 'src/components';
 
-import { AccountsService, ApiError } from 'src/services/openapi';
+import { AccountsService, ApiError, UserProfile } from 'src/services/openapi';
+import { Lens as LensIcon, HelpOutline as HelpIcon } from '@material-ui/icons';
+import { useTheme } from '@material-ui/styles';
+
+const TrustStatus = ({ isTrusted }: { isTrusted: boolean }) => {
+  const theme = useTheme<Theme>();
+
+  return (
+    <Box display="flex" flexDirection="row" flexWrap="wrap" alignItems="center">
+      <Typography>
+        <Box display="flex" gridGap="4px" alignItems="center">
+          Email status:
+          <Box
+            display="inline-flex"
+            flexDirection="row"
+            alignItems="center"
+            gridGap="4px"
+            color={
+              isTrusted
+                ? theme.palette.success.dark
+                : theme.palette.warning.dark
+            }
+            fontWeight="bold"
+          >
+            <LensIcon style={{ fontSize: 16 }} />
+            <Grid item>{isTrusted ? 'Trusted' : 'Non-trusted'}</Grid>
+          </Box>
+        </Box>
+      </Typography>
+
+      <Box display="inline-flex" marginLeft="auto">
+        <Button
+          component={RouterLink}
+          to="/about/trusted_domains"
+          size="small"
+          style={{ fontSize: 13, color: '#777' }}
+          startIcon={<HelpIcon />}
+        >
+          Learn more about trusted domains
+        </Button>
+      </Box>
+    </Box>
+  );
+};
```

Comment:
I slightly dislike the excessive amount of Css Boxes into Css Boxes. Something like this seems to do the same job:

```
      <Typography style={{display: 'flex', alignItems: "center"}}>
        Email status:
        <LensIcon style={{ fontSize: 16, color: statusColor, marginLeft: 8 }} />
        <Box color={statusColor} fontWeight="bold">{isTrusted ? 'Trusted' : 'Non-trusted'}</Box>
      </Typography>
```

From my point of view I love having less code to scroll through, but maybe there are advantages that I fail to see?

Also, I find the layout nicer with a `justifyContent="space-between"` on the top Box instead of the `marginLeft="auto"` for medium screens

Reply:
Thanks for the simplification, after a few iterations my solution was a bit convoluted indeed.
I added your suggestions :+1: